apps-rendering/src/components/liveBlocks.tsx

```diff
@@ -0,0 +1,97 @@
+// ----- Imports ----- //
+
+import { css } from '@emotion/react';
+import { FirstPublished } from '@guardian/common-rendering/src/components/FirstPublished';
+import { LastUpdated } from '@guardian/common-rendering/src/components/LastUpdated';
+import LiveBlockContainer from '@guardian/common-rendering/src/components/liveBlockContainer';
+import { headline, space } from '@guardian/source-foundations';
+import { OptionKind, partition } from '@guardian/types';
+import type { DeadBlog, LiveBlog } from 'item';
+import type { FC } from 'react';
+import { renderAll } from 'renderer';
+
+// ----- Component ----- //
+interface LiveBlocksProps {
+	item: DeadBlog | LiveBlog;
+}
+
+interface BlockTitleProps {
+	title: string;
+}
+
+const BlockTitle: FC<BlockTitleProps> = ({ title }) => {
+	return (
+		<h2
+			css={css`
+				${headline.xxsmall({ fontWeight: 'bold' })}
+				margin-bottom: ${space[2]}px;
+			`}
+		>
+			{title}
+		</h2>
+	);
+};
+
+const LiveBlocks: FC<LiveBlocksProps> = ({ item }) => {
+	return (
+		<div>
+			{/* Accordion? */}
+			{/* Pagination? */}
+			{item.blocks.map((block) => {
+				// TODO: get page number
+				const blockLink = `${1}#block-${block.id}`;
+
+				return (
+					<LiveBlockContainer
+						key={block.id}
+						id={block.id}
+						borderColour={'black'}
+					>
+						<header
+							css={css`
+								padding-right: ${space[3]}px;
+								display: flex;
+								flex-direction: column;
+							`}
+						>
+							{block.firstPublished.kind === OptionKind.Some && (
+								<FirstPublished
+									firstPublished={Number(
+										block.firstPublished.value,
+									)}
+									blockLink={blockLink}
+								/>
+							)}
+							{block.title && <BlockTitle title={block.title} />}
```

Comment:
Is this a guard against empty strings? This is subjective because I know the `&&` is a common JS pattern, but can I suggest making this check explicit for clarity? The `&&` causes a couple of implicit conversions that might be considered harder to reason about 🤔 

```suggestion
							{block.title !== '' ? <BlockTitle title={block.title} /> : null}
```

Reply:
Please always make a suggestion :D In this case I'm not sure, possibly the `&&` pattern is more in-keeping with react? https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

Will think about this one a little more :)

Reply:
I don't think React is opinionated about this. That part of the docs talks more broadly about the fact there are a few different ways to inline conditionals - the heading below the one you linked talks about using ternary for example. It also finishes with this phrase 🙂:

> Just like in JavaScript, it is up to you to choose an appropriate style based on what you and your team consider more readable

It's up to us! 😅 

Actually the last example in the `&&` section demonstrates one of the issues I'm talking about. Doing `a && b` implicitly converts `a` to a boolean, and returns `b` if that boolean is `true`. But if that boolean is `false` it returns `a`. In our code above, if `block.title` is an empty string then you're _rendering an empty string_, you're not rendering nothing.

In modern React I _think_ the visual result is the same in this case, i.e. nothing is inserted into the DOM. However, the recommended way to render nothing is `null` as described by that same docs page. Furthermore, in other circumstances the result may _not_ be nothing inserted into the DOM - as in the example, if this were a number then `0` would get rendered, which may not be what you want to happen.

Reply:
Just to follow up on this - I understand your point and looking through apps-rendering I think this pattern is more widely used. I decided to change to your suggestion :)